content/movie-review-pt1.md

```diff
@@ -0,0 +1,464 @@
+    // Match the returned data struct to what you expect
+    match instruction {
+        NoteInstruction::CreateNote { title, body, id } => {
+            // Execute program code to create a note
+        },
+        NoteInstruction::UpdateNote { title, body, id } => {
+            // Execute program code to update a note
+        },
+        NoteInstruction::DeleteNote { id } => {
+            // Execute program code to delete a note
+        }
+    }
+}
+```
+
+For simple programs where there are only one or two instructions to execute, it may be fine to write the logic inside the match statement. For programs with many different possible instructions to match against, your code will be much more readable if the logic for each instruction is written in a separate function and simply called from inside the `match` statement.
+
+## Program file structure
+
+The [Hello World lesson’s](hello-world-program.md) program was simple enough that it could be confined to one file. But as the complexity of a program grows, it's important to maintain a project structure that remains readable and extensible. This involves encapsulating code into functions, data structures, and functions as we've shown so far. But it also involves grouping related code into separate files.
+
+For example, a good portion of the code we've worked through so far has to do with defining and deserializing instructions. That code should live in its own file rather than be written in the same file as the entry point. By doing so, we would then have 2 files, one with the program entry point and the other with the instruction code:
+
+- **lib.rs**
+- **instruction.rs**
+
+Once you start splitting your program up like this you will need to make sure you register all of the files in one central location. We’ll be doing this in `lib.rs`. **You must register every file in your program like this.**
+
+```rust
+// This would be inside lib.rs
+pub mod instruction;
+```
+
+Additionally, any declarations that you would like to be available through `use` statements in other files will need to be prefaced with the `pub` keyword:
+
+```rust
+pub enum NoteInstruction { ... }
+```
+
+## Demo
+
+For this lesson’s demo, we’ll be building out the first half of the Movie Review program with a focus on deserializing the instruction data. The following lesson will focus on the second half of this program.
+
+### 1. Entry point
+
+We’ll be using [Solana Playground](https://beta.solpg.io/) again to build out this program. Solana Playground saves state in your browser, so everything you did in the previous lesson should still be there. To get started, let's clear everything out from the current `lib.rs` file.
+
+Inside lib.rs, we’re going to bring in the following crates and define where we’d like our entry point to the program to be with the `entrypoint` macro.
+
+```rust
+use solana_program::{
+    entrypoint,
+    entrypoint::ProgramResult,
+    pubkey::Pubkey,
+    msg,
+    account_info::AccountInfo,
```

Comment:
@jamesrp13 fix: trailing comma

ALSO: fwiw, in Hello World Program lesson we list `account_info::AccountInfo` first if we want to be consistent (boring, I know)

Reply:
Probably not gonna change that. Not an area where I think consistency matters